opencti-platform/opencti-graphql/src/modules/workspace/workspace-domain.ts

```diff
@@ -0,0 +1,202 @@
+
+export const findById = (context: AuthContext, user: AuthUser, workspaceId: string): BasicStoreEntityWorkspace => {
+  return storeLoadById(context, user, workspaceId, ENTITY_TYPE_WORKSPACE) as unknown as BasicStoreEntityWorkspace;
+};
+
+export const findAll = (context: AuthContext, user: AuthUser, args: QueryWorkspacesArgs) => {
+  const listArgs = { ...args, adminBypassUserAccess: args.adminBypassUserAccess ?? false };
+  return listEntitiesPaginated<BasicStoreEntityWorkspace>(context, user, [ENTITY_TYPE_WORKSPACE], listArgs);
+};
+
+export const getAuthorizedMembers = async (context: AuthContext, user: AuthUser, workspace: BasicStoreEntityWorkspace): Promise<MemberAccess[]> => {
+  let authorizedMembers: MemberAccess[] = [];
+  if (!workspace.authorized_members?.length) {
+    return authorizedMembers;
+  }
+  if (!validateUserAccessOperation(user, workspace, 'manage-access')) {
+    return authorizedMembers; // return empty if user doesn't have the right access_right
+  }
+  const membersIds = workspace.authorized_members.map((e) => e.id);
+  const args = {
+    connectionFormat: false,
+    first: 100,
+    filters: [{ key: 'internal_id', values: membersIds }],
+  };
+  const members = await findAllMembers(context, user, args);
+  authorizedMembers = workspace.authorized_members.map((am) => {
+    const member = members.find((m) => (m as BasicStoreEntity).id === am.id) as BasicStoreEntity;
+    return { id: am.id, name: member?.name ?? '', entity_type: member?.entity_type ?? '', access_right: am.access_right };
+  });
+  return authorizedMembers;
+};
+
+export const editAuthorizedMembers = async (context: AuthContext, user: AuthUser, workspaceId: string, input: MemberAccessInput[]) => {
+  // validate input (validate access right) and remove duplicates
+  const filteredInput = input.filter((value, index, array) => {
+    return isValidMemberAccessRight(value.access_right) && array.findIndex((e) => e.id === value.id) === index;
+  });
+  if (!filteredInput.some((e) => e.access_right === MEMBER_ACCESS_RIGHT_ADMIN)) {
+    throw FunctionalError('Workspace should have at least one admin');
+  }
+  const authorizedMembersInput = filteredInput.map((e) => {
+    return { id: e.id, access_right: e.access_right };
+  });
+  const patch = { authorized_members: authorizedMembersInput };
+  const { element } = await patchAttribute(context, user, workspaceId, ENTITY_TYPE_WORKSPACE, patch);
+  return notify(BUS_TOPICS[ENTITY_TYPE_WORKSPACE].EDIT_TOPIC, element, user);
+};
+
+export const getCurrentUserAccessRight = async (context: AuthContext, user: AuthUser, workspace: BasicStoreEntityWorkspace) => {
+  return getUserAccessRight(user, workspace);
+};
+
+export const getOwner = async (context: AuthContext, user: AuthUser, workspace: BasicStoreEntityWorkspace) => {
+  const creators = await creatorLoader.load(workspace.creator_id, context, context.user);
+  const owner = creators.length > 0 ? creators[0] : SYSTEM_USER;
+  return owner;
+};
```

Comment:
Indeed, I am not sure if is still work because of the await before the await creatorLoader
It's maybe better to work inside the promise

Need to be test to confirm